flytekit/clis/flyte_cli/main.py

```diff
@@ -1442,5 +1443,43 @@ def register_project(identifier, name, host, insecure):
     _click.echo("Registered project [id: {}, name: {}]".format(identifier, name))
 
 
+@_flyte_cli.command('register-workflow', cls=_FlyteSubCommand)
+@_host_option
+@_insecure_option
+@_click.argument(
+    "closure",
+    type=_click.File("rb"),
+)
+def register_workflow(host, insecure, closure):
+    """
+    Register a new workflow from a serialized workflow-closure protobuf specified in
+    \b
+      https://github.com/lyft/flyteidl/blob/master/protos/flyteidl/core/workflow_closure.proto
+    \b
+      Argument: `closure` Path to the Workflow closure protobuf file that should be registered with flyteadmin
+
+    """
+    _welcome_message()
+
+    c = _workflow_closure_pb2.WorkflowClosure()
```

Comment:
i think a problem here is the order of registration

Reply:
Like what order?

It looks like this script can only register tasks and workflows.  All tasks are registered first and tasks can't depend on one another so that's fine.  Then the workflow is registered, so the order should be fine.  If you start adding in launch plans and stuff, that is where it will get more confusing in terms of ordering.